From 40456c3b7db9be792513e000d1ffb60238002f8d Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Thu, 16 Apr 2015 17:20:20 +0800
Subject: [PATCH 0970/1594] MLK-10670 ARM: dts: imx7d-12x12-arm2: correct
 touch chip's slave address

commit 4cf9e77917e9e9e9b1d7f6bcbe370d4696bbf404 from
git://git.freescale.com/imx/linux-2.6-imx.git

There's an error message when boot kernel.

...
i2c i2c-0: IMX I2C adapter registered
max732x 2-0068: failed reading
i2c i2c-2: Failed to register i2c client max17135 at 0x48 (-16)
i2c i2c-2: of_i2c: Failure registering /soc/aips-bus@30800000/i2c@30a40000/max17135@48
...

This error is introduced by touch driver patch. The reason is that
the TSC2007(touch) has the same I2C slave address as MAX17135 (Eink display
PMIC). This will cause that MAX17135 can not be detected.
The board has to be done with a rework (populate R172), and thus TSC2007's
slave address changes to 0x49. This patch updates the address to 0x49.

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index 53fe47f..8bf760f 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -904,9 +904,9 @@
 	pinctrl-0 = <&pinctrl_i2c3_1>;
 	status = "okay";
 
-	tsc2007@48 {
+	tsc2007@49 {
 		compatible = "ti,tsc2007";
-		reg = <0x48>;
+		reg = <0x49>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_tsc2007>;
 		interrupt-parent = <&gpio1>;
-- 
1.7.5.4

